src/pages/Models/Details/ModelDetails.js

```diff
@@ -45,9 +45,22 @@ const relationTableHeaders = [
   { content: "message" }
 ];
 
-// Temp function to add link to <td> values
-const wrapLink = (href, text) => {
-  return <a href={href}>{text}</a>;
+const generateEntityLink = (entity, href, name) => {
```

Comment:
I noticed that while this does work according to the QA notes, with a few small changes you could make it work for the units as well.

```suggestion
const generateEntityLink = (namespace, href, name) => {
  return (
    <>
      {namespace && (
        <img
          alt=""
          width="24"
          height="24"
          className="entity-icon"
          src={`https://api.jujucharms.com/charmstore/v5/${namespace}/icon.svg`}
        />
      )}
      <a href={href}>{name}</a>
    </>
  );
}
```